### PR TITLE
Extracted job title info from historical data

### PR DESCRIPTION
**Pull request by apuhegde**
Working modular version of job title script.

### CODE REVIEW

File: archive/job_title.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    #standardize title column name
+    body_df.rename(columns={body_df.columns[colIdx]: colname}, inplace=True)
+
+    #remove rows that don't start with any of the following: 1., 1), 1:, Title, Current title, DE, AE, DA, ETL Dev
+    patterns = ["1.", "1)", "1:", "1-", "Title", "Current Title", "Data Engineer", "Data Analyst", "Analytics Engineer", "ETL Developer"]
+    filter_condition = ~body_df[colname].str.lower().str.startswith(tuple(pattern.lower() for pattern in patterns))
+    body_df.loc[filter_condition] = ""
+
+    #remove numbers like "1.", etc. from the beginning of the string
+    body_df[colname] = body_df[colname].str.replace("^[ ]?1[.:)][ ]?","",regex=True)
```
Comment (JPHaus) [inline archive/job_title.py:20]: Great job on the regex!

File: .streamlit/example.secrets.toml
Comment (JPHaus) [inline .streamlit/example.secrets.toml:1]: We need this example file - would you mind adding it back?

File: .gitignore
Comment (JPHaus) [inline .gitignore:1]: Updated the .gitignore to ignore the metadata files/folders.

File: archive/extract_historical_data.py [hunk 2]
```diff
@@ -1,18 +1,69 @@
...
+
+#dir path of input json files
+input_dir = './archive'
+json_files = []
+final_df = pd.DataFrame()
+
+#list all json in input dir
+for file in os.listdir(input_dir):
+    if file.endswith(".json"):
+        json_files.append(os.path.join(input_dir, file))
```
Comment (JPHaus) [inline archive/extract_historical_data.py]:
```suggestion
import glob

json_files = glob.glob(os.path.join(input_dir, '*.json'))
```
This way is completely fine and understandable I just wanted to show an alternate way of getting all of the files in a directory.

**Review by JPHaus — CHANGES_REQUESTED** [review]
I requested removing the metadata files but otherwise the code is a solid contribution!

File: .gitignore
Comment (apuhegde) [reply .gitignore:1]: Sounds good.

File: .streamlit/example.secrets.toml
Comment (apuhegde) [reply .streamlit/example.secrets.toml:1]: Done.

**Comment by suleman1412** [comment]
This looks amazing. Very well done @apuhegde on the regex. I wanted to know if creating a function would be a better way to write the code or would it just take more time to call a function and return a value. For example, we can make a function to replace special characters?

**Review by JPHaus — APPROVED** [review]
Sorry this took so long - thanks again for this great contribution and happy Thanksgiving @apuhegde!

**Pull request merged**